src/bc.F

```diff
@@ -1,6 +1,23 @@
   MODULE bc_module
 
   implicit none
+  private
+
+  !GPU enabled
```

Comment:
why GPU and MPI+GPU need to call different subroutines here?

Reply:
@sjsprecious, To aid debugging, I grouped those boundary update routines into those that were used in the serial code versus those that were used only when MPI was enabled.  Also, I grouped those boundary updates that were GPU enabled versus those that were not.

Reply:
Thanks @johnmauff for the explanation. Based on my understanding, using MPI and GPU or not only requires the activation of some CPP flags. Thus we should be able to use a single subroutine for different run types, which also reduces the risk of introducing inconsistency between different versions. But this may be a separate PR, too.